src/smt/smt_engine.h

```diff
@@ -24,6 +24,7 @@
 #include <string>
 #include <vector>
 
+#include "api/cvc4cpp.h"
```

Comment:
This is an absolute no go and should not be needed. The public header should never be included in the internal code.

Reply:
Agreed, perhaps its for `api::Result` below, you should probably extract the relevant data and make it take only internal data.